google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentSnapshot.java

```diff
@@ -468,6 +468,42 @@ boolean isEmpty() {
     return write;
   }
 
+  /**
+   * Returns the value of the field as an array.
+   *
+   * @param field The path to the field.
+   * @throws IllegalArgumentException if the value is not an array.
+   * @return The value of the field as an array.
+   */
+  @Nullable
+  public <T> List<T> getArray(@Nonnull String field) {
+    if (field == null) {
+      return null;
+    }
+    Object fieldValue = get(field);
+    Preconditions.checkArgument(
```

Comment:
I'm not sure this is the right exception here. The problem is perhaps not the argument.